src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1353,6 +1353,157 @@ def get_webapp_sitecontainer_log(cmd, name, resource_group, container_name, slot
         raise AzureInternalError("Failed to fetch sitecontainer logs. Error: {}".format(str(ex)))
 
 
+def convert_webapp_sitecontainers(cmd, name, resource_group, mode, slot=None):
+    """
+    Convert a webapp between classic and sitecontainers mode.
+
+    :param cmd: CLI command context
+    :param name: Name of the webapp
+    :param resource_group: Resource group of the webapp
+    :param mode: Target mode, either 'classic' or 'sitecontainers'
+    :param slot: Optional deployment slot
+    """
+    client = web_client_factory(cmd.cli_ctx)
+    if mode not in ['classic', 'sitecontainers']:
+        raise InvalidArgumentValueError(
+            "Invalid mode '{}'. Allowed values: classic, sitecontainers.".format(mode)
+        )
+
+    site_config = get_site_configs(cmd, resource_group, name, slot)
+    linux_fx_version = getattr(site_config, "linux_fx_version", None)
+ 
+    if mode == 'sitecontainers':
+        if linux_fx_version and not linux_fx_version.startswith('DOCKER|'):
+            raise ValidationError("Cannot convert to sitecontainers mode as site is not a classic custom container app.")
+        acr_use_managed_identity_creds = getattr(site_config, "acr_use_managed_identity_creds", None)
+        acr_user_managed_identity_id = getattr(site_config, "acr_user_managed_identity_id", None)
+
+        acr_user_name = None
+        acr_user_password = None
+
+        from azure.cli.core.commands.client_factory import get_subscription_id
+        subscription_id = get_subscription_id(cmd.cli_ctx)
+        url = (
+            f"/subscriptions/{subscription_id}/resourceGroups/{resource_group}/"
```

Comment:
If the conversion is performed on a slot, would this endpoint fetch app settings for slot ? or do we need to explicitly handle this?

Reply:
Thanks for noticing this, fixed !!